sip/sip-001-burn-election.md

```diff
@@ -794,8 +808,33 @@ that had a tunable proof score _b_ out of a total sum of _B_ scores receives _b/
 sum(C[N]...C[N+k])_ rewards.
 
 * Batched transactions:  the transaction fees for batched transactions are
-  awarded exclusively to the leader that produced the block.  This ensures that leaders are
-rewarded for processing and validating transactions correctly.
+  distributed to leaders such that leaders are rewarded for processing and
+validating transactions correctly, but in a way that leads to a fair "auction"
+of transaction processing capacity across users.  Specifically, a fraction _F_ of the
+transaction fees will be averaged over _B_ blocks, so a miner will receive _b/B * F *
+sum(T[N]...T[N+k])_ STX in expectation, where:
+
+   * _T[i]_ is the transaction fee for block _i_,
+   * _b_ is the total tunable proof score over _k_ blocks from this miner,
+   * _B_ is the total tunable proof score over _k_ blocks over all miners.
+
+   In addition, the miner will receive _(1 - F) * T[i]_ transaction fees if it mines block
+   _i_, and 0 if it does not.  In other words, the first _F_% of transaction fees are "smoothed"
+   across all miners, and the remaining _(1 - F)%_ are awarded exclusively to
+   the miner.
+
+   An anchored block must be _F_% "full" to receive its transaction
+   fees.  Failure to do so will result in its block not being relayed by the peer network
+   if there are pending transactions.  If there are no pending transactions -- i.e.
```

Comment:
> Failure to do so will result in its block not being relayed by the peer network
   if there are pending transactions.

I'm not sure that this is enforceable as such in our network. In a blockchain where miners are "racing" to produce blocks, honest peers delaying a block-relay will lead to a honest-coalition fork that will eventually eclipse the underfull block. However, in Stacks, this wouldn't be the case, since once leader election has occurred, there is only one valid miner for a given block.

Alternatively, we could say that an honest leader should fork away such underfull blocks.

Reply:
In both the cited paper and here, honest nodes would simply not relay the block if they had sufficiently many transactions in their mempool as to consider this block invalid.

Reply:
Okay, makes sense